src/header/common/strict_transport_security.rs

```diff
@@ -0,0 +1,259 @@
+///
+/// let mut headers = Headers::new();
+///
+/// headers.set(
+///    StrictTransportSecurity(true, 31536000u64)
+/// );
+/// # }
+/// ```
+#[derive(Clone, PartialEq, Debug)]
+pub struct StrictTransportSecurity(pub bool, pub u64);
+
+enum StsDirective {
+    IncludeSubdomains(bool),
+    MaxAge(u64)
+}
+
+struct StsParser {
+    pos: usize,
+    input: Vec<u8>
+}
+
+impl StsParser {
+    fn peek(&self) -> char {
+        *self.input[self.pos..].iter().next().unwrap() as char
+    }
+
+    fn eof(&self) -> bool {
+        self.pos >= self.input.len()
+    }
+
+    fn pop(&mut self) -> char {
+        let c = *self.input[self.pos..].iter().next().unwrap() as char;
+        self.pos += 1;
+        c
+    }
+
+    fn pop_while<F>(&mut self, test: F) -> String where F: Fn(char) -> bool {
+        let mut result = String::new();
+        while !self.eof() && test(self.peek()) {
+            result.push(self.pop());
+        }
+
+        result
+    }
+
+    fn pop_whitespace(&mut self) {
+        self.pop_while(|c: char| c.is_whitespace());
+    }
+
+    fn pop_to_sep(&mut self) {
+        self.pop_while(|c| c != ';');
+
+        if !self.eof() {
+            self.pop();
+        }
+    }
+
+    fn parse_tokens(&mut self) -> String {
+        self.pop_whitespace();
+
+        let result = self.pop_while(|c| match c {
+            '(' | ')' | '<' | '>' | '@' | ',' | ';' | ':' | '\\' |
+            '"' | '/' | '[' | ']' | '?' | '=' | '{' | '}' | ' ' | '\t' => false,
+            _ => true
+        });
+
+        self.pop_whitespace();
+
+        result
+    }
+
+    fn parse_directive_name(&mut self) -> String {
+        self.parse_tokens()
+    }
+
+    fn parse_value(&mut self) -> String {
+        self.parse_tokens()
+    }
+
+    fn parse_quoted_string(&mut self) -> String {
+        let mut result = String::new();
+        self.pop();
+
+        loop {
+            let next = match self.pop() {
+                '\\' => self.pop(),
+                '"' => break,
+                c @ _ => c
+            };
+
+            result.push(next)
+        }
+
+        self.pop_whitespace();
+
+        result
+    }
+
+    fn parse_directive_value(&mut self) -> Option<String> {
+        self.pop_whitespace();
+        if self.eof() { return None }
+
+        match self.peek() {
+            '=' => {
+                self.pop();
+                self.pop_whitespace();
+
+                if self.eof() { return None }
+
+                let result = match self.peek() {
+                    '"' => Some(self.parse_quoted_string()),
+                    _ => Some(self.parse_value())
+                };
+
+                self.pop_to_sep();
+
+                result
+            }
+            _ => {
+                self.pop_to_sep();
+
+                None
+            }
+        }
+    }
+
+    fn parse_directive(&mut self) -> Option<StsDirective> {
+        let directive_name = self.parse_directive_name().to_lowercase();
```

Comment:
I believe this line is the `unstable` feature failing on stable. And since it allocates a new String that gets tossed after the comparison, it'd be better to just compare ignoring case. You can do so using `UniCase`.

```rust
let name = UniCase(self.parse_directive_name());
if name == UniCase("max-age") {

} else if name == UniCase("includeSubdomains") {

} else {

}
```